client-v2/src/main/java/com/clickhouse/client/api/internal/HttpAPIClientHelper.java

```diff
@@ -646,6 +644,56 @@ public static Map<String, String> parseUrlParameters(URL url) {
         return params;
     }
 
+
+    private void correctUserAgentHeader(HttpRequest request, Map<String, Object> requestConfig) {

```

Comment:
I guess it will be called every request why not prepare it once?

Reply:
yes.
we can restrict only changing client name, but not a header. Because we need to alter header value - we have this code. 
We can ignore the case when user completely overrides operation header but it is not how prev. client works. 
I've tried to  minimize work here.